h/static/scripts/directive/annotation-share-dialog.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+var angular = require('angular');
+
+module.exports = function () {
+  return {
+    bindToController: true,
+    controllerAs: 'vm',
+    restrict: 'E',
+    template: require('../../../templates/client/annotation_share_dialog.html'),
+    controller: ['$timeout', '$scope', function ($timeout, $scope) {
+      this.copyToClipboard = function (event) {
+        var $container = angular.element(event.currentTarget).parent();
+        var shareLinkInput = $container.find('input')[0];
+
+        try {
+          shareLinkInput.select();
+          document.execCommand('copy');
+
+          $scope.copyToClipboardMessage = 'Link copied to clipboard!';
+          $scope.copyToClipboardMessageClass = 'annotation-share-dialog-link--success';
+        } catch (ex) {
+          $scope.copyToClipboardMessage = 'Please use Ctrl+C to copy link.';
```

Comment:
Unfortunately this message does not fit in the available space, at least in my testing under Chrome. We also need to take account of the fact that it doesn't make sense on mobile devices.

I can't immediately think of a good short message which is applicable to a reasonable range of desktop and mobile platforms though. "Select and copy to share"?